lib/model/query/datasets.js

```diff
@@ -203,9 +205,13 @@ const getDatasetDiff = (formDefId) => ({ all }) => all(sql`
   .then(reduceBy((acc, { propertyName, isPropertyNew }) => (propertyName ? acc.concat({ name: propertyName, isNew: isPropertyNew }) : acc), [], (row) => `${row.datasetName},${row.isDatasetNew}`))
   .then(r => Object.keys(r).map(k => ({ name: k.split(',')[0], isNew: k.split(',')[1] === 'true', properties: r[k] })));
 
+const getByProjectAndName = (projectId, name) => ({ maybeOne }) => maybeOne(sql`
+SELECT * FROM datasets WHERE "projectId" = ${projectId} AND name = ${name}`)
+  .then(map(construct(Dataset)));
+
 module.exports = {
   createOrMerge,
   getById, getAllByProjectId, getByName,
   getFieldsByFormDefId,
-  getDatasetDiff
+  getDatasetDiff, getByProjectAndName
```

Comment:
Eventually we will want to consolidate/rename one or both of `getByName` and `getByProjectAndName`. In a dataset download resource, we need to use one of these to get a dataset Option that we can call `getOrReject` on.